deeplake/core/chunk/uncompressed_chunk.py

```diff
@@ -25,48 +29,123 @@ def extend_if_has_space(  # type: ignore
                 )
         return self._extend_if_has_space_list(incoming_samples, update_tensor_meta)
 
+    def _extend_if_has_space_text(
+        self,
+        incoming_samples,
+        update_tensor_meta: bool = True,
+        lengths=None,
+    ) -> float:
+        csum = np.cumsum(lengths)
+        min_chunk_size = self.min_chunk_size
+        num_data_bytes = self.num_data_bytes
+        space_left = min_chunk_size - num_data_bytes
+        idx = np.searchsorted(csum, space_left)
```

Comment:
Binary search over cumulative sum of lengths gives the the number of samples that can be accepted by the chunk. Very fast.